google/cloud/storage/internal/async/connection_impl_test.cc

```diff
@@ -399,6 +420,115 @@ TEST_F(AsyncConnectionImplTest, AsyncReadObject) {
   EXPECT_THAT(data.get(), VariantWith<Status>(IsOk()));
 }
 
+TEST_F(AsyncConnectionImplTest, AsyncReadObjectWithTimeout) {
+  AsyncSequencer<bool> sequencer;
+
+  auto mock = std::make_shared<storage::testing::MockStorageStub>();
+  EXPECT_CALL(*mock, AsyncReadObject).WillOnce([&] {
+    auto stream = std::make_unique<MockAsyncObjectMediaStream>();
+    EXPECT_CALL(*stream, Start).WillOnce([&] {
+      return sequencer.PushBack("Start");
+    });
+    EXPECT_CALL(*stream, Read)
+        .WillOnce([&] {
+          return sequencer.PushBack("Read").then([](auto) {
+            google::storage::v2::ReadObjectResponse response;
+            response.mutable_metadata()->set_bucket(
+                "projects/_/buckets/test-bucket");
+            response.mutable_metadata()->set_name("test-object");
+            response.mutable_metadata()->set_size(4096);
+            response.mutable_content_range()->set_start(1024);
+            response.mutable_content_range()->set_end(2048);
+            return absl::make_optional(response);
+          });
+        })
+        .WillOnce([&] {
+          return sequencer.PushBack("Read").then([](auto) {
+            return absl::optional<google::storage::v2::ReadObjectResponse>();
+          });
+        });
+    EXPECT_CALL(*stream, Finish).WillOnce([&] {
+      return sequencer.PushBack("Finish").then([](auto) { return Status{}; });
+    });
+    return std::unique_ptr<AsyncReadObjectStream>(std::move(stream));
+  });
+
+  auto mock_cq = std::make_shared<MockCompletionQueueImpl>();
+  // We will configure the connection to use 1 second timeouts.
+  EXPECT_CALL(*mock_cq, MakeRelativeTimer(
+                            std::chrono::nanoseconds(std::chrono::seconds(1))))
```

Comment:
We don't need to cast to nanoseconds first, right?  `MakeRelativeTimer()` will do it for us.

Reply:
This is setting a mock expectation, and `std::chrono::nanoseconds(std::chrono::seconds(1))` is just the matcher.  If I use `std::chrono::seconds()` I get:

```
[build] /usr/local/google/home/coryan/google-cloud-cpp/google/cloud/storage/internal/async/connection_impl_test.cc:459:42: error: no matching function for call to ‘google::cloud::v2_23::testing_util::MockCompletionQueueImpl::gmock_MakeRelativeTimer(std::chrono::seconds)’
[build]   459 |   EXPECT_CALL(*mock_cq, MakeRelativeTimer(
[build]       |                                          ^
```

I guess if we had a matcher that could compare `std::chrono::duration` with different `Period` parameters I could use that instead, but we don't.

Reply:
I tried with `Eq(std::chrono::seconds(1))`, no luck either 🤷

Reply:
I see.  Thanks.